src/VirtoCommerce.ElasticSearch8.Data/Services/ElasticSearch8Provider.cs

```diff
@@ -100,10 +100,69 @@ ILogger<ElasticSearch8Provider> logger
             }
         }
 
-        public async Task<SearchResponse> SearchAsync(string documentType, SearchModule.Core.Model.SearchRequest request)
+        public Task<SearchResponse> SearchAsync(string documentType, SearchModule.Core.Model.SearchRequest request)
         {
             CheckClientCreated();
 
+            return InternalSearchAsync(documentType, request);
```

Comment:
It's not required to create separate protected Internal methods, just add virtual keyword to existing ones.